### PR TITLE
PICARD-1130: Add a post save hook

### PR DESCRIPTION
**Pull request by zas**
- register_file_post_save_processor()
- run_file_post_save_processors() is called in File._save_and_rename() just before returning

# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): [PICARD-1130](https://tickets.metabrainz.org/browse/PICARD-1130)

# Solution

# Action

### CODE REVIEW

File: picard/file.py [hunk 1]
```diff
@@ -285,6 +289,7 @@ def _save_and_rename(self, old_filename, metadata):
         # Save cover art images
         if config.setting["save_images_to_files"]:
             self._save_images(os.path.dirname(new_filename), metadata)
+        run_file_post_save_processors(self, new_filename)
```
Comment (mineo) [inline picard/file.py]: I'm not sure if this is the correct place for this - https://github.com/metabrainz/picard/blob/a1a79658bf58d150c927e7ff03b96c8e938d0ae6/picard/file.py#L213 calls `_saving_finished` *after* `_save_and_rename` and `_saving_finished` still does a lot of work in the non-error case, including updating `self.filename` and some metadata. If we want to merge this in its current form, the documentation needs to clearly list which updates are guaranteed to have happened at the point the post-save-processor runs and which are still not done. In particular the not-yet-updated filename of the `File` object in both the `filename` attribute and the metadata is super problematic for the automatic-submission-to-acousticbrainz usecase.

Comment (zas) [reply picard/file.py]: The file is actually saved in `_save_and_rename`, but i agree it makes more sense to call the hook in `_saving_finished`, and just pass file object.

Comment (zas) [reply picard/file.py]: I did that in 5e93e1529fb09e, does it look better ?

File: picard/file.py [hunk 2]
```diff
@@ -768,3 +775,22 @@ def discnumber(self):
...
+
+
+_file_post_save_processors = PluginFunctions(label='file_post_save_processors')
+
+
+def register_file_post_save_processor(function, priority=PluginPriority.NORMAL):
+    """Registers file saved processor.
+
+    Args:
+        function: function to call after save, it will be passed file object and new filename
```
Comment (mineo) [inline picard/file.py]: Only a file object.

**Pull request merged**